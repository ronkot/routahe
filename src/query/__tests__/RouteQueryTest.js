import moment from 'moment'
import RouteQuery from '../RouteQuery'

describe('HSL-API', function() {
  jasmine.DEFAULT_TIMEOUT_INTERVAL = 50000

  const nextWednesday = moment().day(10).hour(15).minute(13)

  it('should be able to get route', async() => {
    const routes = await new RouteQuery().fetch({
      from: { label: 'Kottaraistie 1a, Helsinki', lon: 25.005448, lat: 60.263282 },
      to:  { label: 'Kottaraistie 4a, Helsinki', lon: 25.004373, lat: 60.263418 },
      date: nextWednesday,
      arriveBy: false
    })

    const filteredRoutes = routes.map(route => removeRouteTimeAttributes(route))

    expect(filteredRoutes[0]).toMatchSnapshot()  // First route is walk-only
  })

  it('should be able to get route with arriveBy', async() => {
    const routes = await new RouteQuery().fetch({
      from: { label: 'Kottaraistie 1a, Helsinki', lon: 25.005448, lat: 60.263282 },
      to:  { label: 'Kottaraistie 4a, Helsinki', lon: 25.004373, lat: 60.263418 },
      date: nextWednesday,
      arriveBy: true
    })

    const filteredRoutes = routes.map(route => removeRouteTimeAttributes(route))
    expect(filteredRoutes[0]).toMatchSnapshot()
  })

  function removeRouteTimeAttributes(route) {
    route.legs = route.legs.map(leg => removeLegTimeAttributes(leg))
    return route
  }

  function removeLegTimeAttributes(leg) {
    const {duration, distance, mode, from, to, route} = leg
    return {duration, distance, mode, from, to, route}
  }
})