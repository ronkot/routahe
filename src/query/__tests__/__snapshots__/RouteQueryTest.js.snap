exports[`HSL-API should be able to get route 1`] = `
Object {
  "duration": 56,
  "legs": Array [
    Object {
      "distance": 55.397000000000006,
      "duration": 56,
      "from": Object {
        "name": "Origin",
      },
      "mode": "WALK",
      "route": null,
      "to": Object {
        "name": "Destination",
      },
    },
  ],
}
`;

exports[`HSL-API should be able to get route with arriveBy 1`] = `
Object {
  "duration": 56,
  "legs": Array [
    Object {
      "distance": 55.397000000000006,
      "duration": 56,
      "from": Object {
        "name": "Origin",
      },
      "mode": "WALK",
      "route": null,
      "to": Object {
        "name": "Destination",
      },
    },
  ],
}
`;
